src/core/resource_reader.rs

```diff
@@ -60,6 +70,30 @@ impl Direct {
     pub fn init(runtime: TargetRuntime) -> Self {
         Self { runtime }
     }
+
+    // makes http request with passed headers.
+    pub async fn get<T: ToString + Send>(
+        &self,
+        path: T,
+        headers: Option<BTreeMap<String, String>>,
+    ) -> anyhow::Result<serde_json::Value> {
+        let url = Url::parse(&path.to_string())?;
+        let mut request = reqwest::Request::new(reqwest::Method::GET, url);
+
+        if let Some(headers_inner) = headers {
+            let mut header_map = HeaderMap::new();
+            for (key, value) in headers_inner {
+                let header_name = HeaderName::try_from(key)?;
+                let header_value = HeaderValue::try_from(value)?;
+                header_map.insert(header_name, header_value);
+            }
+            *request.headers_mut() = header_map;
+        }
+
+        let response = self.runtime.http.execute(request).await?;
+        let response: serde_json::Value = serde_json::from_slice(&response.body)?;
+        Ok(response)
+    }
 }
```

Comment:
I think it's a better idea to merge the `get` with the `read` instead of having two separate implementations. The current implementation doesn't allow usage of the `Cached` reader. We might need to change the `Reader` trait so that it can accept additional information. Here is what I propose: 



```rust

// Create a new type called source
enum Source {
    File(String),
    Request(Request)
}

// Reader will take a source as input
trait Reader {
    async fn read<T: Into<Source>>(&self, file: T) -> anyhow::Result<FileRead>
}

// So that it works for our current solutions
impl Source for String {
     // ... 
}

// So that it support any arbitary request
// We use this for sampling and config generation
impl Source for Request {
   // ...
}


```